// generate-random-color/lib/index.js
'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.hex = exports.rgba = exports.rgb = exports.hsla = exports.hsl = void 0;
var hsl_1 = require('./color-types/hsl');
var hsla_1 = require('./color-types/hsla');
var rgb_1 = require('./color-types/rgb');
var rgba_1 = require('./color-types/rgba');
var hex_1 = require('./color-types/hex');
var hsl = function (input) {
  var _a, _b, _c, _d, _e, _f;
  var base = {
    h: {
      min:
        ((_a = input === null || input === void 0 ? void 0 : input.h) === null || _a === void 0 ? void 0 : _a.min) || 0,
      max:
        ((_b = input === null || input === void 0 ? void 0 : input.h) === null || _b === void 0 ? void 0 : _b.max) ||
        361,
    },
    s: {
      min:
        ((_c = input === null || input === void 0 ? void 0 : input.s) === null || _c === void 0 ? void 0 : _c.min) || 0,
      max:
        ((_d = input === null || input === void 0 ? void 0 : input.s) === null || _d === void 0 ? void 0 : _d.max) ||
        101,
    },
    l: {
      min:
        ((_e = input === null || input === void 0 ? void 0 : input.l) === null || _e === void 0 ? void 0 : _e.min) || 0,
      max:
        ((_f = input === null || input === void 0 ? void 0 : input.l) === null || _f === void 0 ? void 0 : _f.max) ||
        101,
    },
  };
  return hsl_1.default(base);
};
exports.hsl = hsl;
var hsla = function (input) {
  var _a, _b, _c, _d, _e, _f, _g, _h;
  var base = {
    h: {
      min:
        ((_a = input === null || input === void 0 ? void 0 : input.h) === null || _a === void 0 ? void 0 : _a.min) || 0,
      max:
        ((_b = input === null || input === void 0 ? void 0 : input.h) === null || _b === void 0 ? void 0 : _b.max) ||
        361,
    },
    s: {
      min:
        ((_c = input === null || input === void 0 ? void 0 : input.s) === null || _c === void 0 ? void 0 : _c.min) || 0,
      max:
        ((_d = input === null || input === void 0 ? void 0 : input.s) === null || _d === void 0 ? void 0 : _d.max) ||
        101,
    },
    l: {
      min:
        ((_e = input === null || input === void 0 ? void 0 : input.l) === null || _e === void 0 ? void 0 : _e.min) || 0,
      max:
        ((_f = input === null || input === void 0 ? void 0 : input.l) === null || _f === void 0 ? void 0 : _f.max) ||
        101,
    },
    a: {
      min:
        ((_g = input === null || input === void 0 ? void 0 : input.a) === null || _g === void 0 ? void 0 : _g.min) || 0,
      max:
        ((_h = input === null || input === void 0 ? void 0 : input.a) === null || _h === void 0 ? void 0 : _h.max) || 1,
    },
  };
  return hsla_1.default(base);
};
exports.hsla = hsla;
var rgb = function (input) {
  var _a, _b, _c, _d, _e, _f;
  var base = {
    r: {
      min:
        ((_a = input === null || input === void 0 ? void 0 : input.r) === null || _a === void 0 ? void 0 : _a.min) || 0,
      max:
        ((_b = input === null || input === void 0 ? void 0 : input.r) === null || _b === void 0 ? void 0 : _b.max) ||
        256,
    },
    g: {
      min:
        ((_c = input === null || input === void 0 ? void 0 : input.g) === null || _c === void 0 ? void 0 : _c.min) || 0,
      max:
        ((_d = input === null || input === void 0 ? void 0 : input.g) === null || _d === void 0 ? void 0 : _d.max) ||
        256,
    },
    b: {
      min:
        ((_e = input === null || input === void 0 ? void 0 : input.b) === null || _e === void 0 ? void 0 : _e.min) || 0,
      max:
        ((_f = input === null || input === void 0 ? void 0 : input.b) === null || _f === void 0 ? void 0 : _f.max) ||
        256,
    },
  };
  return rgb_1.default(base);
};
exports.rgb = rgb;
var rgba = function (input) {
  var _a, _b, _c, _d, _e, _f, _g, _h;
  var base = {
    r: {
      min:
        ((_a = input === null || input === void 0 ? void 0 : input.r) === null || _a === void 0 ? void 0 : _a.min) || 0,
      max:
        ((_b = input === null || input === void 0 ? void 0 : input.r) === null || _b === void 0 ? void 0 : _b.max) ||
        255,
    },
    g: {
      min:
        ((_c = input === null || input === void 0 ? void 0 : input.g) === null || _c === void 0 ? void 0 : _c.min) || 0,
      max:
        ((_d = input === null || input === void 0 ? void 0 : input.g) === null || _d === void 0 ? void 0 : _d.max) ||
        255,
    },
    b: {
      min:
        ((_e = input === null || input === void 0 ? void 0 : input.b) === null || _e === void 0 ? void 0 : _e.min) || 0,
      max:
        ((_f = input === null || input === void 0 ? void 0 : input.b) === null || _f === void 0 ? void 0 : _f.max) ||
        255,
    },
    a: {
      min:
        ((_g = input === null || input === void 0 ? void 0 : input.a) === null || _g === void 0 ? void 0 : _g.min) || 0,
      max:
        ((_h = input === null || input === void 0 ? void 0 : input.a) === null || _h === void 0 ? void 0 : _h.max) || 1,
    },
  };
  return rgba_1.default(base);
};
exports.rgba = rgba;
var hex = function (input) {
  if (input) {
    return hex_1.default(input);
  } else {
    return hex_1.default();
  }
};
exports.hex = hex;
